Automattic/jetpack
# Add partner subsidiary id to upgrade URLs.

#### Changes proposed in this Pull Request:

* Upgrade URLs will now globally include a subsidiary ID (if it exists). This value will come from a blog option called `jetpack_partner_subsidiary_id` that our hosting partners can set on the site themselves.

### Why?

Once we provision a Jetpack site for a hosting partner we customize the URLs we show users at the time of upgrading so they can follow the upgrade flow proposed by the partner that provisioned the site. 

Each of our partners gets a provisioning key that they use to provision sites and we use to keep track of stats and billing. This change allows a partner to configure upgrade URLs for many of their subsidiaries while using a single partner key (which allows us to keep stats and billing simple).

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This is a new feature which is part of the Partner Configuration project.
* Master Thread: parMIr-l2-p2
* The first partner that will use this feature is EuroDNS. 

#### Testing instructions:






* Checkout this PR.
* Run the following code snippet on your test site:

```php
<?php
function subsidiary_filter( $subsidiary ) {
	return 'from-filter-' . $subsidiary;
}

update_option( 'jetpack_partner_subsidiary_id', 'some-subsidiary-123', true );

add_filter( 'jetpack_partner_subsidiary_id', 'subsidiary_filter' );

```

> Tip: The easiest way I found to do this without running any CLI commands is to install the Hello Dolly plugin and use the plugin editor to paste the above code snippet right at the end of the `wp-content/plugins/hello-dolly/hello.php` file. Now, activate the plugin and the option will be set.

* Got to any page where Jetpack offers you an upgrade button. The plans page is a good choice.
* Inspect the upgrade link and you should see `&subsidiaryId=from-filter-some-subsidiary-123` appended at the end. The "from-filter-" string is added by the filter and the rest of the string is added by the option so if you see both, everything works.

In order to run the unit tests, a new test suite has been created:

```bash
yarn docker:phpunit --testsuite=partner-subsidiary
```

#### Proposed changelog entry for your changes:

* Add support for hosting partners to configure a subsidiary id to upgrade URLs.
